File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
object Raindrops {

    private val factors = mapOf(
        3 to "Pling",
        5 to "Plang",
        7 to "Plong",
    )

    fun convert(n: Int) = buildString {
        factors.filterKeys { n % it == 0 }.values.forEach(this::append)

        if (isBlank()) {
            append(n)
        }
    }
}

File: swift/raindrops/Sources/Raindrops/Raindrops.swift
private let factors = [(3, "Pling"), (5, "Plang"), (7, "Plong")]

func raindrops(_ number: Int) -> String {
  let result =
    factors
    .filter { number.isMultiple(of: $0.0) }
    .map { $0.1 }
    .joined()

  return result.isEmpty ? String(number) : result
}
